R/field.R

```diff
@@ -37,6 +37,14 @@ merge.roxy_field <- function(x, y, ...) {
   roxy_field_simple(x$field, c(x$values, y$values))
 }
 
+#' @export
+merge.roxy_field_param <- function(x, y, ...) {
+  stopifnot(identical(class(x), class(y)))
+  # When parameters appear in both x and y, keep values from y
```

Comment:
I would've thought that the first definition would win, not the last — I think that would be more consistent with `@inheritParams`?

Reply:
@hadley , this order is not the order in which the tags appear in the roxygen comments, but it is the order in which the code executes, and this order is first `inherit_params()` and then `inherit_dot_params()`. So if `inherit_params()` has added "..." already, then we want `inherit_dot_params()` to overwrite this. I have added a comment to explain this.

Reply:
In that case, I don't think you should do it in this more general `merge()` method; I think you should just inline this code directly in `merge.roxy_field_inherit_dot_params()`.

Reply:
The way I think about it is that the params field should never have any duplicates. If the code tries to set the same param value twice, it should be the later one that overwrites the earlier one, because that is how assignments in code generally work. With the code written this way, if someone else in the future again writes code that merges stuff into the params field (as the author of inhert_dot_params() did), duplicates will automatically be avoided. 

Of course I would be happy to follow your recommendations, but I am not sure how to do that elegantly. The function merge.roxy_field_inherit_dot_params() is never called with the params field, so how can it remove the old "..." slot from it that inherit_params() had added.

Reply:
I'm just saying that this code should go in `merge.roxy_field_inherit_dot_params()`; in complicated packages like roxygen2 you want to make the simplest possible change to avoid unintended consequences.

Or are you saying that both merge methods need to be modified in order for your approach to work?

If that's the case, I think we need to avoid the `...` duplication at some other place, as making the last parameter definition win seems like a major change to me.

Reply:
I agree with keeping bigfixes simple. I also agree with keeping bugfixes local. However in this case we could describe the bug more generally as: "the params field sometimes contains duplicates". I agree that currently the duplicate happens to always be "...", but any other duplicate would be just as bad. So I think it is appropriate to fix the bug in a way that will always avoid duplicates, i.e., by not allowing `merge.roxy_field_params()` to create duplicates. It makes the code more robust against possibly someone in the future again writing code that produces duplicates.

I think it is also the simplest way to fix the bug. Currently both `inherit_params()` and `inherit_dot_params()` independently of each other decide what they want to add to the params field and each calls `merge.roxy_field_params()` to do that (via the `add_simple_field()` method), one after the other. When `inherit_params()` is running, it does not know yet that `inherit_dot_params()` will run after it and will add "...", so it adds its own "..." value to the params field by calling `merge.roxy_field_params()`. When `inherit_dot_params()` runs afterwards, it needs to overwrite this with its own "..." value and it does that by also calling `merge.roxy_field_params()`. The only problem is that currently `merge.roxy_field_params()` creates a duplicate "...".

If we don't want `merge.roxy_field_params()` to always avoid duplicates, we could tell `inherit_dot_params()` to check in the `RoxyTopic` object whether it already has a "..." value for its params field and delete that value before calling `merge.roxy_field_params()`. That however seems a bit more hacky to me, especially as there is no "remove" method currently provided by the `roxy_field` class. However I could of course add one. Would you prefer that route?

Her is one other option: the `add_simple_field()` and `add_field()` methods already have an argument called `overwrite = FALSE` that will make it overwrite the existing field. How about giving it another argument `overwrite_value = FALSE` that it would pass on to `merge()` and that would then allow `merge()` to overwrite existing values in the field, rather than creating duplicates. Then `inherit_dot_params()` could call `add_simple_field()` with `overwrite_value = TRUE`.